### PR TITLE
OEM-IBM: BIOS attribute support for LINUX KVM

### PR DESCRIPTION
**Pull request by vkaverap**
This commit adds new BIOS attributes to support Linux KVM

### CODE REVIEW

File: oem/ibm/configurations/bios/enum_attrs.json [hunk 1]
```diff
@@ -573,6 +573,28 @@
             "default_values": ["MTU1500"],
             "helpText": "Specifies the maximum frame size (maximum transmission unit, MTU). The value is in terms of bytes per packet size.",
             "displayName": "Maximum Frame Size"
+        },
+        {
+            "attribute_name": "pvm_default_os_type",
```
Comment (jaypadath) [inline oem/ibm/configurations/bios/enum_attrs.json:578]: We already have a existing pvm_default_os_type why you want to create a new one here ?

File: oem/ibm/configurations/bios/integer_attrs.json [hunk 2]
```diff
@@ -285,6 +285,25 @@
...
             "displayName": "Target Port"
+        },
+        {
+            "attribute_name": "pvm_linux_kvm_percentage",
+            "lower_bound": 0,
+            "upper_bound": 1000,
+            "scalar_increment": 1,
+            "default_value": 0,
+            "helpText": "Specifies the percentage of available system memory that will be reserved for the management of KVM guests. The percentage is specified to the 10th of a percent.",
+            "displayName": "Percentage of Available System Memory Reserved for KVM Guest Management"
```
Comment (jaypadath) [inline oem/ibm/configurations/bios/integer_attrs.json:296]: I remember we were talking on removing "Percentage of Available" from the Display name ?

File: oem/ibm/configurations/bios/integer_attrs.json [hunk 3]
```diff
@@ -285,6 +285,25 @@
...
+        },
+        {
+            "attribute_name": "pvm_linux_kvm_percentage_current",
+            "lower_bound": 0,
+            "upper_bound": 1000,
+            "scalar_increment": 1,
+            "default_value": 0,
+            "readOnly": true,
+            "helpText": "Specifies the percentage of available system memory that will be reserved for the management of KVM guests. Do not set this attribute directly; set pvm_linux_kvm_percentage instead.",
+            "displayName": "Percentage of Available System Memory Reserved for KVM Guest Management (current)"
```
Comment (jaypadath) [inline oem/ibm/configurations/bios/integer_attrs.json:306]: I remember we were talking on removing "Percentage of Available" from the Display name ?

**Review by jaypadath — CHANGES_REQUESTED** [review]
Please elaborate the commit message and add the tested section too.

File: oem/ibm/configurations/bios/integer_attrs.json (hunk 2 above)
Comment (vkaverap) [reply oem/ibm/configurations/bios/integer_attrs.json:296]: i think that was not yet decided

**Comment by riyadixitagra** [comment]
The CI checks are failing. Please fix them

> Please elaborate the commit message and add the tested section too.
+1

File: oem/ibm/configurations/bios/enum_attrs.json (hunk 1 above)
Comment (riyadixitagra) [reply oem/ibm/configurations/bios/enum_attrs.json:578]: this commit is only changing - possibleValues, defaultValues, helpText, displayName for pvm_default_os_type. Why is code from line 576 to 578 being added as a part of your commit

**Review by riyadixitagra — COMMENTED** [review]
The CI checks are failing. Please fix them.

**Review by jaypadath — COMMENTED** [review]
Also please change commit title from "PLDM:" to "oem-ibm: "

File: oem/ibm/configurations/bios/enum_attrs.json (hunk 1 above)
Comment (vkaverap) [reply oem/ibm/configurations/bios/enum_attrs.json:578]: removed the pvm_default_os_type as it is already defined

File: oem/ibm/configurations/bios/integer_attrs.json (hunk 2 above)
Comment (vkaverap) [reply oem/ibm/configurations/bios/integer_attrs.json:296]: removed

File: oem/ibm/configurations/bios/enum_attrs.json (hunk 1 above)
Comment (riyadixitagra) [reply oem/ibm/configurations/bios/enum_attrs.json]: In the initial commit change you changes some values for pvm_default_os_type, why are we removing all of it? Is it not required ? As per the attribute doc some values need change, shouldn't this be part of this commit?

**Review by riyadixitagra — COMMENTED** [review]
Please address the pvm_default_os_type change

**Comment by vkaverap** [comment]
> Please address the pvm_default_os_type change

the changes are already there in file no need to add

File: oem/ibm/configurations/bios/enum_attrs.json [hunk 4]
```diff
@@ -573,6 +573,21 @@
             "default_values": ["MTU1500"],
             "helpText": "Specifies the maximum frame size (maximum transmission unit, MTU). The value is in terms of bytes per packet size.",
             "displayName": "Maximum Frame Size"
+        },
+        {
+            "attribute_name": "pvm_linux_kvm_memory",
+            "possible_values": ["Automatic", "Custom"],
+            "default_values": ["Automatic"],
+            "helpText": "Controls whether the system or user specified percentage of available system memory will be reserved for the management of KVM guests.",
```
Comment (riyadixitagra) [inline oem/ibm/configurations/bios/enum_attrs.json:581]: Isn't it better to write the complete helptext that is mentioned in attribute doc as it enhances understanding

File: oem/ibm/configurations/bios/enum_attrs.json [hunk 5]
```diff
@@ -573,6 +573,21 @@
...
+            "displayName": "Memory setting for KVM Guest Management"
+        },
+        {
+            "attribute_name": "pvm_linux_kvm_memory_current",
+            "possible_values": ["Automatic", "Custom"],
+            "default_values": ["Automatic"],
+            "helpText": "Controls whether the system or user specified percentage of available system memory will be reserved for the management of KVM guests. This attribute is set by the BMC. Set pvm_linux_kvm_memory instead.",
```
Comment (riyadixitagra) [inline oem/ibm/configurations/bios/enum_attrs.json:588]: Isn't it better to write the complete helptext that is mentioned in attribute doc as it enhances understanding

Comment (vkaverap) [reply oem/ibm/configurations/bios/enum_attrs.json:588]: no we dont need to , that was discussed in our meetings with GUI and host

File: oem/ibm/configurations/bios/enum_attrs.json (hunk 4 above)
Comment (vkaverap) [reply oem/ibm/configurations/bios/enum_attrs.json:581]: no we dont need to , that was discussed in our meetings with GUI and host

Comment (vkaverap) [reply oem/ibm/configurations/bios/enum_attrs.json:581]: there is no need to explain definition of custom and automatic in help text

File: oem/ibm/configurations/bios/enum_attrs.json (hunk 5 above)
Comment (vkaverap) [reply oem/ibm/configurations/bios/enum_attrs.json:588]: there is no need to explain definition of custom and automatic in help text

**Review by jaypadath — COMMENTED** [review]
In the commit message, please change "OEM-IBM" to "oem-ibm".  Add more details on the commit message description as apart from Linux KVM, we are adding memory setting changes too.  Also add the Tested By field.

**Review by jaypadath — APPROVED** [review]

**Review by riyadixitagra — APPROVED** [review]

**Pull request merged**